### PR TITLE
feat(fgs/function): support config max instance number

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
1. Support new parameter 'max_instance_num' and related test.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support new parameter 'max_instance_num' and related test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFunction_strategy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFunction_strategy -timeout 360m -parallel 4
=== RUN   TestAccFunction_strategy
=== PAUSE TestAccFunction_strategy
=== CONT  TestAccFunction_strategy
--- PASS: TestAccFunction_strategy (53.46s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       53.578s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_strategy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_strategy -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_strategy
=== PAUSE TestAccFgsV2Function_strategy
=== CONT  TestAccFgsV2Function_strategy
--- PASS: TestAccFgsV2Function_strategy (31.33s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       31.408s
```

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 1]
```diff
@@ -326,6 +340,17 @@ func resourceFgsFunctionV2Create(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if strNum, ok := d.GetOk("max_instance_num"); ok {
+		maxInstanceNum, err := strconv.Atoi(strNum.(string))
+		if err != nil {
```
Comment (ShiChangkuo) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: it's better to convert the value to int before creating the FGS function.

Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: done. I bring this type conversion (validation) into the schema definition (ValidateFunc).

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_strategy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_strategy -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_strategy
=== PAUSE TestAccFgsV2Function_strategy
=== CONT  TestAccFgsV2Function_strategy
--- PASS: TestAccFgsV2Function_strategy (28.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       28.756s
```

**Pull request merged**